preprocessing/regex_sentence_segmentation.py

```diff
@@ -1,32 +1,31 @@
 import re
 
+RE_SPACES = re.compile("[ \n]{1,}")
 
-def sent_segmentation(text):
+def sentence_segmentation(text):
     """Return a list of sentences from the text."""
     
-    text = text.replace("\n", " ")
-    while "  " in text:
-        text = text.replace("  ", " ")
+    text = RE_SPACES.sub(" ", text)
 
     sentences = []
     sent = ""
 
-    for i in range(len(text)-2):
+    for i in range(len(text) - 2):
 
         # sentence can't start with end quotations marks
-        if len(sent) == 0 and text[i] == "\"“":
-            break
-
-        # check if not end of text
-        # check if char is interpunction
-        # check if previous char is not capital (no segmentation on abreviations)
-        # check if char after space is capital
-        # check beginning of next sentence
-        elif (i == len(text)-1 or i != len(text)-3 or i != len(text)-2 or i != 0) \
-            and (re.search(r"[.!?;:]", text[i]) \
-            and re.search(r"[^A-Z]", text[i-1]) \
-            and (re.search(r"[A-Z'“\"]", text[i+2])) \
-            or (re.search(r"[\"”]", text[i+1]) and re.search(r"[A-Z'“\"]", text[i+3]))):
+        if len(sent) == 0 and (text[i] == "\"" or text[i] == "“"):
```

Comment:
This is perfectly fine, Python programmer would beautify it even further:
`if not sent and text[i] in '“"'`